test/mux.sh

```diff
@@ -0,0 +1,353 @@
+    local is_testproj=$2
+    local no_container_vals=$3
+    local mismatch_namespace=${4:-0}
+    local no_project_tag=${5:-0}
+
+    local uuid_es=$( uuidgen )
+    local uuid_es_ops=$( uuidgen )
+
+    add_test_message $uuid_es
+    logger -i -p local6.info -t $uuid_es_ops $uuid_es_ops
+
+    local rc=0
+
+    os::log::debug "is_testproj $is_testproj no_container_vals $no_container_vals ===================================="
+
+    espod=$es_pod
+    if [ $is_testproj -eq 1 -a $no_container_vals -eq 0 ]; then
+        # kibana logs with project.testproj tag and given container/pod values
+        myfield="message"
+        mymessage=$uuid_es
+        myproject=project.testproj
+    else
+        # kibana logs with kibana container/pod values
+        myfield="message"
+        mymessage=$uuid_es
+        myproject=project.logging
+    fi
+    os::cmd::try_until_success "curl_es $espod /${myproject}.*/_count?q=${myfield}:$mymessage | get_count_from_json | grep -q 1" "$(( 10*minute ))"
+
+    if [ $is_testproj -eq 1 ]; then
+        # other logs with project.testproj tag
+        myfield="MESSAGE"
+        myproject=project.testproj
+        espod=$es_pod
+    elif [ $no_project_tag -eq 1 ]; then
+        myfield="MESSAGE"
+        myproject=project.mux-undefined
+        espod=$es_pod
+    else
+        myfield="message"
+        myproject=".operations"
+        espod=$es_ops_pod
+    fi
+    mymessage=$uuid_es_ops
+    os::cmd::try_until_success "curl_es $espod /${myproject}.*/_count?q=${myfield}:$mymessage | get_count_from_json | grep -q 1" "$(( 10*minute ))"
+}
+
+reset_ES_HOST() {
+    oc set env dc logging-mux $1 $2
+    os::cmd::try_until_failure "oc get pod $muxpod"
+    os::cmd::try_until_text "oc get pods -l component=mux" "^logging-mux-.* Running "
+    muxpod=$( get_running_pod mux )
+}
+
+cleanup() {
+    local return_code="$?"
+    set +e
+    if [ $return_code = 0 ] ; then
+        mycmd=os::log::info
+    else
+        mycmd=os::log::error
+        os::log::debug "$( oc projects )"
+        os::log::debug "$( oc get pods )"
+        if [ -n "$fpod" ]; then
+            os::log::debug "$( oc get configmap/logging-fluentd -o yaml )"
+            os::log::debug "$( oc exec $fpod -- ls /etc/fluent/configs.d/openshift )"
+            os::log::debug "$( oc exec $fpod -- ls /etc/fluent/configs.d/user )"
+        fi
+        if [ "$muxpod" != "" ]; then
+            os::log::debug "$( oc logs $muxpod )"
+            os::log::debug "$( oc get configmap/logging-mux -o yaml )"
+            os::log::debug "$( oc exec $muxpod -- ls /etc/fluent/configs.d/openshift )"
+            os::log::debug "$( oc exec $muxpod -- ls /etc/fluent/configs.d/user )"
+        fi
+    fi
+    $mycmd mux test finished at $( date )
+    # dump the pod before we restart it
+    if [ -n "${fpod:-}" ] ; then
+        oc logs $fpod > $ARTIFACT_DIR/$fpod.log 2>&1
+    fi
+    os::log::debug "$( oc label node --all logging-infra-fluentd- 2>&1 || : )"
```

Comment:
This is in the cleanup code - it may be that something failed when the node was unlabeled, so if the node is already unlabeled that's ok